PyRDF/backend/Utils.py

```diff
@@ -1,10 +1,41 @@
-import ROOT
-import os
 import logging
+import multiprocessing
+import os
+import threading
+
+try:
+    import queue
+except ImportError:
+    import Queue as queue
+
+import ROOT
 
 logger = logging.getLogger(__name__)
 
 
+def run_graphs(proxies, numthreads=multiprocessing.cpu_count()):
+    """
+    Trigger multiple RDF graphs through multithreading.
+    """
+    ROOT.EnableThreadSafety()
+
+    q = queue.Queue()
+
+    for proxy in proxies:
+        q.put(proxy)
+
+    def trigger_loop(queue_):
+        while True:
+            queue_.get().GetValue()
+            queue_.task_done()
+
+    for _ in range(numthreads):
+        worker = threading.Thread(target=trigger_loop, args=(q,), daemon=True)
```

Comment:
So what you are doing here is creating multiple Python threads, where each thread will trigger the loop of one of the result proxies. If the selected backend is Spark, this will result in three Python threads submitting one job each to the same Spark context, is this correct?

Reply:
Correct, this is the intended behaviour as suggested in the Spark docs

Reply:
I guess there is no issue when multiple Python threads submit to the same Spark context object, since anyway there is no real concurrency? In practice, what will happen is that the Spark context will do the submissions of the jobs one after the other, and from there the Spark scheduler will take over?

Reply:
The SparkContext will receive the different jobs submission virtually at the same time (depending on the timing of the threads), then the Spark scheduler will treat them as different jobs so they will run on the cluster concurrently depending on the cluster configuration. The default for a standalone Spark cluster is that any job can receive all resources and the scheduling policy is FIFO, but that really doesn't depend on the PyRDF level

Reply:
There is no real concurrency on the driver machine, no operations on data will be done inside the local threads